app/src/main/java/chat/rocket/android/dagger/module/AppModule.kt

```diff
@@ -5,18 +5,35 @@ import android.arch.persistence.room.Room
 import android.content.Context
 import chat.rocket.android.BuildConfig
 import chat.rocket.android.app.RocketChatDatabase
+import chat.rocket.android.authentication.infraestructure.AuthTokenRepository
 import chat.rocket.android.server.infraestructure.ServerDao
 import chat.rocket.android.util.TimberLogger
 import chat.rocket.common.util.PlatformLogger
+import chat.rocket.core.RocketChatClient
 import dagger.Module
 import dagger.Provides
+import okhttp3.HttpUrl
 import okhttp3.OkHttpClient
 import okhttp3.logging.HttpLoggingInterceptor
 import javax.inject.Singleton
 
 @Module
 class AppModule {
 
+    @Provides
+    @Singleton
+    fun provideRocketChatClient(okHttpClient: OkHttpClient, repository: AuthTokenRepository, logger: PlatformLogger): RocketChatClient {
```

Comment:
We can't provide the RocketChatClient at the application level because we need the server URL when creating an instance...

The idea is to create a RocketChatClientProvider that could cache clients for different servers (for some time, so we don't need to recreate one every time we change the server)

Reply:
I added the provider because it is better than creating a `RocketChatClient` in each presenter as before. About the server url I think that we should don't force the user to specify a specific server when creating a `RocketChatClient` (provide the `AuthTokenRepository` and `PlatformLogger` is another thing that isn't good IMHO). Think about others SDK like AWS or Firebase: it request only the user ID, regions...
I believe that having something like `rocketchatclient.connect("server_url")` would be nice.